pep-9999.rst

```diff
@@ -46,38 +46,46 @@ Rationale
 =========
 
 This PEP proposes basing the standard library support for reading TOML on the
-third party library ``tomli`` [2]_.
+third party library ``tomli``
+(`github.com/hukkin/tomli <https://github.com/hukkin/tomli>`_).
 
 Many projects have recently switched to using ``tomli``, for example, ``pip``,
 ``build``, ``pytest``, ``mypy``, ``black``, ``flit``, ``coverage``,
 ``setuptools-scm``, ``cibuildwheel``.
 
 ``tomli`` is actively maintained and well-tested. ``tomli`` is about 800 lines
 of code with 100% test coverage and passes all tests in a test suite `proposed as
-the official TOML compliance test suite <https://github.com/toml-lang/compliance/pull/8>`,
-as well as `the more established BurntSushi/toml-test suite <https://github.com/BurntSushi/toml-test>`.
+the official TOML compliance test suite <https://github.com/toml-lang/compliance/pull/8>`_,
+as well as `the more established BurntSushi/toml-test suite <https://github.com/BurntSushi/toml-test>`_.
 
 
 Specification
 =============
 
+A new module ``tomllib`` with the following functions will be added:
+
 .. code-block::
 
    def load(fp: SupportsRead[bytes], /, *, parse_float: Callable[[str], Any] = float) -> dict[str, Any]: ...
    def loads(s: str, /, *, parse_float: Callable[[str], Any] = float) -> dict[str, Any]: ...
 
-``tomllib.load`` deserializes a ``.read()``-supporting binary file containing a
-TOML document to a Python object.
+``tomllib.load`` deserializes a binary file containing a
+TOML document to a Python dict.
+The ``fp`` argument must have a ``read()`` method with the same API as
+``io.RawIOBase.read()``.
 
-``tomllib.loads`` deserializes a str instance containing a TOML document to a
-Python object.
+``tomllib.loads`` deserializes a str instance containing a TOML document
+to a Python dict.
 
 ``parse_float`` is a function that takes a string and returns a float, as with ``json.load``.
-For example, ``decimal.Decimal`` in cases where precision is important.
+For example, ``decimal.Decimal`` can be used in cases where precision is important.
+
+The returned object contains only basic Python objects (``str``, ``int``, ``bool``, ``float``,
+``datetime.{datetime,date,time}``, ``list``, ``dict`` with string keys).
```

Comment:
If you pass a custom `parse_float`, this won't necessarily be true... right?

Reply:
That is correct!